ZSSRforKernelGAN/ZSSR.py

```diff
@@ -137,34 +145,12 @@ def set_disc_loss(self, D, DiscLoss):
     def run(self):
         # Run gradually on all scale factors (if only one jump then this loop only happens once)
         print('*' * 60 + '\nSTARTED ZSSR on: \"%s\"...' % self.input_img_path)
-        for self.sf_ind, (sf, self.kernel) in enumerate(zip(self.conf.scale_factors, self.kernels)):
-            # Relative_sf (used when base change is enabled. this is when input is the output of some previous scale)
-            sf = [sf, sf] if np.isscalar(sf) else sf
-            self.sf = np.array(sf) / np.array(self.base_sf)
-
-            self.output_shape = np.uint(np.ceil(np.array(self.input.shape[0:2]) * sf))
-
-            # Initialize network
-            self.init_sess(init_weights=self.conf.init_net_for_each_sf)
-
-            # Train the network
-            self.train()
-
-            # Use augmented outputs and back projection to enhance result. Also save the result.
-            post_processed_output = self.final_test()
 
-            # Keep the results for the next scale factors SR to use as dataset
-            self.hr_fathers_sources.append(post_processed_output)
+        # Train the network
+        self.train()
 
-            # append a corresponding map loss
-            if self.conf.grad_based_loss_map:
-                self.loss_map_sources.append(create_loss_map(im=post_processed_output))
-            else:
-                self.loss_map_sources.append(np.ones_like(post_processed_output))
-
-            # In some cases, the current output becomes the new input. If indicated and if this is the right scale to
-            # become the new base input. all of these conditions are checked inside the function.
-            self.base_change()
```

Comment:
same question of both the loss map sources and base change.
why can we remove it?

Reply:
again this function begins with the line :
if len(self.conf.base_change_sfs) < self.base_ind + 1:
            return

the default of self.conf.base_change_sfs is 0 so for one scale factor the above statement is aways true